src/Sentry/Scope.cs

```diff
@@ -180,15 +185,52 @@ public string? TransactionName
         }
     }
 
-    private ITransactionTracer? _transaction;
+    /// <summary>
+    /// <para>
+    /// Most of the properties on the Scope should have the same affinity as the Scope... For example, when using a
+    /// GlobalScopeStackContainer, anything you store on the scope will be applied to all events that get sent to Sentry
+    /// (no matter which thread they are sent from).
+    /// </para>
+    /// <para>
+    /// Transactions are an exception, however. We don't want spans from threads created on the UI thread to be added as
+    /// children of Transactions/Spans that get created on the background thread, or vice versa. As such,
+    /// Scope.Transaction is always stored as an AsyncLocal, regardless of the ScopeStackContainer implementation.
+    /// </para>
+    /// <para>
+    /// See https://github.com/getsentry/sentry-dotnet/issues/3590 for more information.
```

Comment:
cc @getsentry/team-mobile 

Should we have this behavior on all mobile SDKs? I think global mode and transactions is suboptimal across the board, right?

Reply:
Yeah I think Scope.Transaction should be AsyncLocal always/everywhere. I can't think of a scenario where it will get us into trouble (famous last words...)